### PR TITLE
Manage subscription page unsubscribes from hidden lists [MAILPOET-2537]

### PR DESCRIPTION
**Pull request by amine-mp**
[MAILPOET-2537]

[MAILPOET-2537]: https://mailpoet.atlassian.net/browse/MAILPOET-2537

### CODE REVIEW

File: lib/Subscription/Manage.php [hunk 1]
```diff
@@ -48,6 +54,19 @@ function onSave() {
     $this->url_helper->redirectBack();
   }
 
+  private function addHiddenSegments(Subscriber $subscriber, array $data) {
+    $shown_segments = $this->settings->get('subscription.segments');
+    if (empty($shown_segments)) return $data;
+    if (empty($data['segments'])) $data['segments'] = [];
```
Comment (JanJakes) [inline lib/Subscription/Manage.php]: What does this line do? The `$data` variable must be an array therefore `empty($data)` already is `[]`.

File: lib/Subscription/Manage.php [hunk 2]
```diff
@@ -48,6 +54,19 @@ function onSave() {
     $this->url_helper->redirectBack();
   }
 
+  private function addHiddenSegments(Subscriber $subscriber, array $data) {
+    $shown_segments = $this->settings->get('subscription.segments');
+    if (empty($shown_segments)) return $data;
```
Comment (JanJakes) [inline lib/Subscription/Manage.php]: We don't have a sniff for this yet but the coding style should be:
```php
if (empty($shown_segments)) {
  return $data;
}
```

File: lib/Subscription/Manage.php [hunk 3]
```diff
@@ -48,6 +54,19 @@ function onSave() {
...
+    if (empty($data['segments'])) $data['segments'] = [];
+    $subscriber->withSubscriptions();
+    foreach ($subscriber->subscriptions as $subscription) {
+      if ($subscription['status'] === Subscriber::STATUS_SUBSCRIBED && !in_array($subscription['segment_id'], $shown_segments)) {
+        $data['segments'][] = $subscription['segment_id'];
+      }
+    }
```
Comment (JanJakes) [inline lib/Subscription/Manage.php]: This is quite a hard to read logic and there's no comment to explain it. But I don't like it anyway - it seems we take data from user's form and then add some stuff to it from the DB (and only specific subscription status). Why don't we do naturally, the other way round? I think we should first fetch all subscription data from the DB, then apply user changes on it, then save it. That would be a more straightforward and safe way to do it.

File: lib/Subscription/Manage.php (hunk 1 above)
Comment (amine-mp) [reply lib/Subscription/Manage.php]: I am checking and initializing `$data['segments']` not `$data`

**Pull request merged**